doc/tutorials/introduction/env_reference/env_reference.markdown

```diff
@@ -0,0 +1,342 @@
+
+##### Levels:
+* `0`, `O`, `OFF`, `S`, `SILENT`, `DISABLE`, `DISABLED`
+* `F`, `FATAL`
+* `E`, `ERROR`
+* `W`, `WARNING`, `WARN`, `WARNINGS`
+* `I`, `INFO`
+* `D`, `DEBUG`
+* `V`, `VERBOSE`
+
+
+### core/parallel_for
+| name | type | default | description |
+|------|------|---------|-------------|
+| :star: OPENCV_FOR_THREADS_NUM | num | 0 | set number of threads |
+| OPENCV_THREAD_POOL_ACTIVE_WAIT_PAUSE_LIMIT | num | 16 | tune pthreads parallel_for backend |
+| OPENCV_THREAD_POOL_ACTIVE_WAIT_WORKER | num | 2000 | tune pthreads parallel_for backend |
+| OPENCV_THREAD_POOL_ACTIVE_WAIT_MAIN | num | 10000 | tune pthreads parallel_for backend |
+| OPENCV_THREAD_POOL_ACTIVE_WAIT_THREADS_LIMIT | num | 0 | tune pthreads parallel_for backend |
+| OPENCV_FOR_OPENMP_DYNAMIC_DISABLE | bool | false | use single OpenMP thread |
+
+
+### backends
+OPENCV_LEGACY_WAITKEY
+Some modules have multiple available backends, following variables allow choosing specific backend or changing default priorities in which backends will be probed (e.g. when opening a video file).
+
+| name | type | default | description |
+|------|------|---------|-------------|
+| OPENCV_PARALLEL_BACKEND | string | | choose specific paralel_for backend (one of `TBB`, `ONETBB`, `OPENMP`) |
+| OPENCV_PARALLEL_PRIORITY_${NAME} | num | | set backend priority, default is 1000 |
+| OPENCV_PARALLEL_PRIORITY_LIST | string, `,`-separated | | list of backends in priority order |
+| OPENCV_UI_BACKEND | string | | choose highgui backend for window rendering (one of `GTK`, `GTK3`, `GTK2`, `QT`, `WIN32`) |
+| OPENCV_UI_PRIORITY_${NAME} | num | | set highgui backend priority, default is 1000 | |
+| OPENCV_UI_PRIORITY_LIST | string, `,`-separated | | list of hioghgui backends in priority order |
+| OPENCV_VIDEOIO_PRIORITY_${NAME} | num | | set videoio backend priority, default is 1000 | |
+| OPENCV_VIDEOIO_PRIORITY_LIST | string, `,`-separated | | list of videoio backends in priority order |
+
+
+### plugins
+Some external dependencies can be detached into a dynamic library, which will be loaded at runtime (plugin). Following variables allow changing default search locations and naming pattern for these plugins.
+| name | type | default | description |
+|------|------|---------|-------------|
+| OPENCV_CORE_PLUGIN_PATH | paths | | directories to search for _core_ plugins |
+| OPENCV_CORE_PARALLEL_PLUGIN_${NAME} | string, glob | | parallel_for plugin library name (glob), e.g. default for TBB is "opencv_core_parallel_tbb*.so" |
+| OPENCV_DNN_PLUGIN_PATH | paths | | directories to search for _dnn_ plugins |
+| OPENCV_DNN_PLUGIN_${NAME} | string, glob | | parallel_for plugin library name (glob), e.g. default for TBB is "opencv_core_parallel_tbb*.so" |
+| OPENCV_CORE_PLUGIN_PATH - for highgui plugins (YES it is CORE) |
+| OPENCV_UI_PLUGIN_${NAME} |
+| OPENCV_VIDEOIO_PLUGIN_PATH |
+| OPENCV_VIDEOIO_PLUGIN_${NAME} |
+
+### OpenCL
+| name | type | default | description |
```

Comment:
Table formatting is broken: https://pullrequest.opencv.org/buildbot/export/pr/23968/docs/d6/dea/tutorial_env_reference.html